String.php

```diff
@@ -767,6 +767,143 @@ public static function getCharDirection ( $char ) {
     }
 
     /**
+     * Get the number of column positions of a wide-character.
+     *
+     * This is a PHP implementation of wcwidth() and wcswidth() (defined in IEEE
+     * Std 1002.1-2001) for Unicode, by Markus Kuhn. Please, see
+     * http://www.cl.cam.ac.uk/~mgk25/ucs/wcwidth.c.
+     *
+     * The wcwidth(wc) function shall either return 0 (if wc is a null
+     * wide-character code), or return the number of column positions to be
+     * occupied by the wide-character code wc, or return -1 (if wc does not
+     * correspond to a printable wide-character code).
+     *
+     * @access  public
+     * @param   string  $char    Character.
+     * @return  int
+     */
+    public static function getCharWidth ( $char ) {
+
+        static $combining = [
+            [0x0300,  0x036f],  [0x0483,  0x0486],  [0x0488,  0x0489],
```

Comment:
Maybe `\p{Mn}` is better, but you're right.

Quoting http://pcre.org/current/doc/html/pcre2pattern.html#uniextseq:
>  M     Mark
>  Mc    Spacing mark
>  Me    Enclosing mark
>  Mn    Non-spacing mark

We are targeting non-spacing mark here, so it's `Mn`. Are you agree?